http_2.rflx

```diff
@@ -0,0 +1,230 @@
+					if Settings_Identifier = SETTINGS_MAX_FRAME_SIZE
+				then Settings_Value_Generic
+					if (Settings_Identifier = SETTINGS_HEADER_TABLE_SIZE or
+						Settings_Identifier = SETTINGS_MAX_CONCURRENT_STREAMS or
+						Settings_Identifier = SETTINGS_MAX_HEADER_LIST_SIZE);
+			Settings_Value_Enable_Push : Settings_Value_Enable_Push
+				then null;
+			Settings_Value_Initial_Window_Size : Settings_Value_Initial_Window_Size
+				then null;
+			Settings_Value_Max_Frame_Size : Settings_Value_Max_Frame_Size
+				then null;
+			Settings_Value_Generic : Settings_Value_Generic
+				then null;
+			Pad_Length : Pad_Length
+				then Data
+					with Size => (Payload_Length - (Pad_Length + 1)) * 8 	-- 1 -> Pad_Length'Length
+					if Frame_Type = DATA and Payload_Length >= Pad_Length + 1
+				then Exclusive_Flag
+					if Frame_Type = HEADERS and Flag_Priority = True
+				then Header_Block_Fragment
+					with Size => Payload_Length * 8
+					if Frame_Type = HEADERS and Flag_Priority = False
+				then Reserved_Bit_2
+					if Frame_Type = PUSH_PROMISE;
+			Reserved_Bit_2 : Zero_Bit
+				then Promised_Stream_ID
+					if Frame_Type = PUSH_PROMISE
+				then Last_Stream_ID 
+					if Frame_Type = GOAWAY
+				then Window_Size_Increment
+					if Frame_Type = WINDOW_UPDATE;
+			Promised_Stream_ID : Stream_ID
+				then Header_Block_Fragment
+					with Size => (Payload_Length - (Pad_Length + 5)) * 8
+					if Flag_Padded = True and Payload_Length >= Pad_Length + 5
+				then Header_Block_Fragment
+					with Size => (Payload_Length - 4) * 8
+					if Flag_Padded = False and Payload_Length >= 4;
+			Last_Stream_ID : Stream_ID
+				then Error_Code;
+			Exclusive_Flag : Boolean
+				then Stream_Dependency;
+			Stream_Dependency : Stream_ID
+				then Weight;
+			Weight : Weight
+				then Header_Block_Fragment
+					with Size => (Payload_Length - (Pad_Length + 6)) * 8
+					if Frame_Type = HEADERS and Flag_Padded = True and Payload_Length >= Pad_Length + 6
+				then Header_Block_Fragment
+					with Size => (Payload_Length - 5) * 8				 -- Sub prev field lengths of frame type
+					if Frame_Type = HEADERS and Flag_Padded = False and Payload_Length >= 5
+				then null
+					if Frame_Type = PRIORITY;
+			Header_Block_Fragment : Opaque
+				then Padding
+					with Size => Pad_Length * 8
+					if (Frame_Type = HEADERS or Frame_Type = PUSH_PROMISE) and Flag_Padded = True
+				then null
+					if Frame_Type = CONTINUATION or
+					 	((Frame_Type = PUSH_PROMISE or
+					 	Frame_Type = HEADERS) and Flag_Padded = False);
+			Data : Opaque
+				then Padding
+					with Size => Pad_Length * 8
+					if Frame_Type = DATA and Flag_Padded = True
+				then null
+					if (Frame_Type = DATA and Flag_Padded = False or Frame_Type = PING);
+			Padding : Opaque
+				then null;
+			Error_Code : Error_Code
+				then null
+					if Frame_Type = RST_STREAM
+				then null
+					if Frame_Type = GOAWAY and Payload_Length <= (Error_Code'Size + Last_Stream_ID'Size + Reserved_Bit_2'Size)
```

Comment:
The edges cannot be merged, as an error is thrown in that case. I think this is because on a path where `Frame_Type = RST_STREAM`, the fields `Last_Stream_ID` and `Reserved_Bit_2` are not present.

```
http_2.rflx:216:133: model: error: undefined variable "Last_Stream_ID"
http_2.rflx:216:155: model: error: undefined variable "Reserved_Bit_2"
http_2.rflx:216:44: model: info: on path Payload_Length -> Frame_Type -> Flag_End_Stream_ACK -> Bit_2_Unused -> Flag_End_Headers -> Flag_Padded -> Bit_5_Unused -> Flag_Priority -> Bit_7_Unused -> Bit_8_Unused -> Reserved_Bit_1 -> Stream_Identifier -> Error_Code -> Final
```

Reply:
This happens when using a disjunction instead of the two edges? This would be a bug that needs to be fixed. If so, could you create a reproducer and open a ticket for RecordFlux, please.

Reply:
You are right. Paths with contradictions are filtered out before the type checking is done. If both edges are merged, the condition is fulfillable for the path where `Last_Stream_ID` and `Reserved_Bit_2` are not present, but the type checking detects the missing fields. We would need to split the expression before checking their satisfiability and doing type checking. I think it wouldn't be easy to handle such cases reliably, while the drawback of needing two links to the same target is quite small. So I tend to leaving it as it is. @senier What's your opinion?

Reply:
Nevertheless, creating an issue for RecordFlux to document that limitation would still be sensible.

Reply:
IMHO this limitation is rather unexpected and mysterious to users and we should try to address it in Componolit/RecordFlux#499. We should give users the freedom to express the formal specification most similar to the original spec. If that involves disjunctions of the above form, this should be possible.

Reply:
Maybe you are right. I have labeled it as bug and moved it to "future", as I don't think it is that important that we fix it immediately.